app/core/worker.py

```diff
@@ -188,3 +241,220 @@ async def remove_product_from_team(db: Session, customer_id: str, product_id: st
         db.rollback()
         logger.error(f"Error removing product from team: {str(e)}")
         raise e
+
+async def monitor_teams(db: Session):
+    """
+    Daily monitoring task for teams that:
+    1. Posts age metrics for teams (since creation for teams without products, since last payment for teams with products)
+    2. Sends notifications for teams approaching expiration (25-30 days)
+    3. Posts metrics for expired teams (>30 days)
+    4. Monitors key spend and notifies if approaching limits
+    """
+    logger.info("Monitoring teams")
+    try:
```

Comment:
nit: could conisder breaking some of this up with helper functions just for readability